Sources/TuistSigning/SecurityController.swift

```diff
@@ -16,13 +16,16 @@ final class SecurityController: SecurityControlling {
     }
 
     func importCertificate(_ certificate: Certificate, keychainPath: AbsolutePath) throws {
-        if try !certificateExists(at: certificate.publicKey, keychainPath: keychainPath) {
+        if try !certificateExists(certificate, keychainPath: keychainPath) {
             try importToKeychain(at: certificate.publicKey, keychainPath: keychainPath)
             logger.debug("Imported certificate at \(certificate.publicKey.pathString)")
         }
-        if try !keyExists(at: certificate.privateKey, keychainPath: keychainPath) {
+        do {
+            // found no way to check for the presence of a private key in the keychain, but fortunately keychain takes care of duplicate private keys on its own
             try importToKeychain(at: certificate.privateKey, keychainPath: keychainPath)
             logger.debug("Imported certificate private key at \(certificate.privateKey.pathString)")
+        } catch {
+            logger.warning("Importing certificate private key at \(certificate.privateKey.pathString) failed")
```

Comment:
I'd fail here - also why have you decided not to bubble up error from `importToKeychain`?

If we decide to throw a new error, I'd try to inspect the error we receive and present a better error message when we can, so user can act on that error (in this case it'd be hard for them to tell what exactly went wrong)